### ISSUE TITLE
Battlelog Exception: Request Timed Out

### ISSUE THREAD

**Issue by Prophet731** [AdKats/AdKats#771]
Guys over at BP4U reported their layers going offline and when looking at plugin logs this error is being spammed a lot. I believe this is causing their layers to crash.

```
[23:52:33] [AdKats] EXCEPTION-6600-D-HandleException-BattlelogComm41: [Error while parsing player stats data.][System.Net.WebException: The request timed out
  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x00000] in <filename unknown>:0
  at System.Net.HttpWebRequest.GetResponse () [0x00000] in <filename unknown>:0
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request) [0x00000] in <filename unknown>:0
  at System.Net.WebClient.ReadAll (System.Net.WebRequest request, System.Object userToken) [0x00000] in <filename unknown>:0
  at System.Net.WebClient.DownloadDataCore (System.Uri address, System.Object userToken) [0x00000] in <filename unknown>:0 ]
```

**Comment by BP4U** [comment]
yes...after switching off adkats all layers are running again without problems

**Comment by BP4U** [comment]
disabling hacker-checker didn't change anything.

**Comment by ColColonCleaner** [comment]
run /pupdate on your instances and let me know if anything changes.

**Comment by BP4U** [comment]
I get the error "unable to download plugin update".

**Comment by BP4U** [comment]
push

its quiet urgent cause nothing is running anymore
pls help

**Comment by ColColonCleaner** [comment]
Did you try manually updating the plugin?

**Comment by BP4U** [comment]
Our adkats is on latest stable version 6.6.0.0
Shall we update to 6.6.7.0?

**Comment by ColColonCleaner** [comment]
Yes. Let me know if that fixes things.

**Comment by BP4U** [comment]
unfortunately that didn't change anything,

[17:40:49 93] [AdKats] EXCEPTION-6670-D-FetchPlayerBattlelogInformation-BattlelogComm49: [Error while parsing player stats data.][System.Net.WebException: The request timed out
  at System.Net.HttpWebRequest.EndGetResponse (IAsyncResult asyncResult) [0x00000] in <filename unknown>:0 
  at System.Net.HttpWebRequest.GetResponse () [0x00000] in <filename unknown>:0 
  at System.Net.WebClient.GetWebResponse (System.Net.WebRequest request) [0x00000] in <filename unknown>:0 
  at System.Net.WebClient.ReadAll (System.Net.WebRequest request, System.Object userToken) [0x00000] in <filename unknown>:0 
  at System.Net.WebClient.DownloadDataCore (System.Uri address, System.Object userToken) [0x00000] in <filename unknown>:0 ]

**Comment by ColColonCleaner** [comment]
Very strange. Are you running insane limits? If so, check to see if that plugin is having issues connecting with battlelog from your layer. The only thing I can think of is it's being blocked from that connection.

What country are your layers in?

**Comment by BP4U** [comment]
nope, no insanelimits. Here is a screenshot of all the stuff running atm: http://fs2.directupload.net/images/150504/48daf8md.png
The layers are running on a linux-root in germany.

**Comment by BP4U** [comment]
ok...so after we installed all layers on my local machine only on one server layer an error message occur @hackercheacker

[18:00:07 73] [AdKats] EXCEPTION-6671-D-FetchPlayerStatInformation-HackerChecker38: [Error while parsing player stats data.][System.Net.WebException: Die Anfrage wurde abgebrochen: Timeout für Vorgang überschritten.
   bei System.Net.WebClient.DownloadDataInternal(Uri address, WebRequest& request)
   bei System.Net.WebClient.DownloadString(Uri address)
   bei PRoConEvents.AdKats.FetchPlayerStatInformation(AdKatsPlayer aPlayer) in e:\Procon\layer_08\layer_08\Plugins\BF4\Temp\k2hfrqz7.0.cs:Zeile 36918.]

we have started hacker-checker again and we will see if problems occur again

**Comment by ColColonCleaner** [comment]
That's why updating to the latest version was important. All attempts at fetching stat information were moved into the hacker checker, instead of being done for everyone. If you disable it do you not see that issue anymore?

**Comment by BP4U** [comment]
well...after we disabled it we did not run it long enough to see...because we switched it on again...no errors so far

**Comment by BP4U** [comment]
after 12 hours i can say that it runs stable, also with hacker cheacker activated

**Comment by ColColonCleaner** [comment]
But...we didn't change anything in the plugin...

Looks like the issue was either with battlelog or your layer. The two exceptions you mentioned above are for concurrent modification and invalid token, they do not affect operation.

Any idea what happened/changed with your layer?

**Comment by BP4U** [comment]
to be honest, i gor no idea what could have happened... Today we will try putting the layers back on the rootserver, let's see what happens then.

**Comment by BP4U** [comment]
today we set up fresh layers on our root and it happens again on one of five layers

[03:15:51 21] [AdKats] EXCEPTION-6670-D-FetchPlayerStatInformation-HackerChecker69: [Error while parsing player stats data.][System.NullReferenceException: Object reference not set to an instance of an object
  at PRoConEvents.AdKats.FetchPlayerStatInformation (PRoConEvents.AdKatsPlayer aPlayer) [0x00000] in <filename unknown>:0 ]

**Comment by ColColonCleaner** [comment]
Please /pupdate to the latest version and turn on plugin debugging on your layer.

**Comment by ColColonCleaner** [comment]
Closed due to it being a duplicate of #783
